src/leetCodeSolutionProvider.ts

```diff
@@ -63,6 +45,27 @@ class LeetCodeSolutionProvider implements Disposable {
         }
     }
 
+    private addMdDefaultHighlight(): void {
+        // The @types typedef of `highlight` is wrong, which should return a string.
+        // tslint:disable-next-line:typedef
+        const highlight = this.markdown.options.highlight as (code: string, lang?: string) => string;
+        this.markdown.options.highlight = (code: string, lang?: string): string => {
+            return highlight(code, lang || this.solution.lang);
+        };
+    }
+
+    private addMdImageUrlCompletion(): void {
```

Comment:
What is this method for? Looks like this method could be moved into the  MarkDownEngine.

Reply:
The host name is context-dependent (markdownEngine.ts#L292):
```typescript
imageSrc[1] = `https://discuss.leetcode.com${imageSrc[1]}`; // only valid in Solution webview
```
If I can manage to find a way to insert host name as a context value when rendering the markdown, maybe it can be moved into the MarkdownEngine.